tools/src/main/scala/scala/scalanative/interflow/MergeProcessor.scala

```diff
@@ -75,7 +75,7 @@ final class MergeProcessor(insts: Array[Inst],
         val mergeEmitted = mutable.Map.empty[Op, Val]
         val newEscapes   = mutable.Set.empty[Addr]
 
-        def mergePhi(values: Seq[Val]): Val = {
+        def mergePhi(values: Seq[Val], bound: Option[Type] = None): Val = {
```

Comment:
Consider removing the default value, so that it's painfully obvious at call site when we use `None` (which is the dangerous value, so it should be visible):
```suggestion
        def mergePhi(values: Seq[Val], bound: Option[Type]): Val = {
```